Dh 9

Dyrk84 opened the pull request.

Dyrk84 committed:
DH-9 add Main.java

Dyrk84 committed:
DH-9
add gitignore
add fightfeature

Dyrk84 committed:
DH-9
dice upgraded
champion values renamed
new methods in main

gergely-szanto-epam committed:
DH-9 fix instantiations; extend pom with java version

Dyrk84 committed:
DH-9
dice upgraded
int numOfDices added to characters
new start() method in class Fight

Dyrk84 committed:
DH-9
test added

gergelyszanto commented on src/main/java/hu/dungeonhunter/Main.java:
```diff
@@ -0,0 +1,28 @@
+            Scanner scanner = new Scanner(System.in);
+            System.out.println("\nChooose one of the following actions:");
+            System.out.println("1. Attack");
+            int chosenNumber = scanner.nextInt();
+            if (chosenNumber == 1) {
```
Please add input validation.

gergelyszanto commented on src/main/java/hu/dungeonhunter/characters/Monsters.java:
```diff
@@ -0,0 +1,37 @@
+    private final int monsterMaxDamage;
+    private int numOfDices;
+
+
```
Remove empty line

gergelyszanto commented on src/main/java/hu/dungeonhunter/characters/Monsters.java:
```diff
@@ -0,0 +1,37 @@
+        this.monsterHP = 20;
+        this.monsterMaxDamage = 6;
+        this.numOfDices = 1;
+
```
Remove empty line

gergelyszanto commented on src/main/java/hu/dungeonhunter/tools/Fight.java:
```diff
@@ -0,0 +1,50 @@
+                monster.getNumOfDices() + "-" + monster.getNumOfDices() * monster.getMonsterMaxDamage() +
+                " damage.");
+    }
+
```
Remove empty line

gergelyszanto commented on src/main/java/hu/dungeonhunter/tools/Fight.java:
```diff
@@ -0,0 +1,50 @@
+        System.out.print("Monster attack: ");
+        champion.setChampionHP(champion.getChampionHP() - monster.monsterDamage());
+        System.out.println("Champion have now " + champion.getChampionHP() + " hit points");
+
```
gameOver should be checked at both monsterAttack and championAttack. If champion kills the monster then monster shouldn't attack the champion.

gergelyszanto commented on src/test/java/hu/dungeonhunter/TestDice.java:
```diff
@@ -0,0 +1,31 @@
+        int result = Dice.rollDice(10,1);
+        assertThat(result).as("A kockadobás eredménye nem 1-10!").isBetween(1, 10);
+    }
+
```
Remove empty line

gergelyszanto commented on src/test/java/hu/dungeonhunter/TestDice.java:
```diff
@@ -0,0 +1,31 @@
+package hu.dungeonhunter;
+
+import hu.dungeonhunter.tools.Dice;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.junit.Test;
+
+import java.util.Random;
```
Remove empty lines between imports

gergely-szanto-epam replied:
You can remove Random import, it is not used

gergelyszanto commented on src/main/java/hu/dungeonhunter/tools/Fight.java:
```diff
@@ -0,0 +1,50 @@
+    }
+
+    public void start() {
+        System.out.println("Your champion have " + champion.getChampionHP() + " HP and can do " +
+                champion.getNumOfDices() + "-" + champion.getNumOfDices() * champion.getChampionMaxDamage() +
+                " damage.");
```
You could move the starting HP and max damage values to the class initializer method for both the champion and the monster and start method could be removed. So whenever you instantiate a monster (or champion), the porgram will automatically print out their stats.

Dyrk84 replied:
I need help in this!

gergely-szanto-epam replied:
See my comment at Champion.java

Dyrk84 replied:
Thank you! Its helped!

gergelyszanto commented on src/main/java/hu/dungeonhunter/characters/Champion.java:
```diff
@@ -0,0 +1,35 @@
+package hu.dungeonhunter.characters;
+
```
Remove empty line

gergelyszanto commented on src/main/java/hu/dungeonhunter/Main.java:
```diff
@@ -0,0 +1,28 @@
+package hu.dungeonhunter;
+
```
Remove empty line

gergely-szanto-epam commented on src/test/java/hu/dungeonhunter/TestDice.java:
```diff
@@ -0,0 +1,31 @@
+    }
+
+    @Test
+    public void testRollDiceRandomDice() {
```
Testing is not random, rename test method to testRollDice10d1

gergely-szanto-epam commented on src/test/java/hu/dungeonhunter/TestDice.java:
```diff
@@ -0,0 +1,31 @@
+    }
+
+    @Test
+    public void testRollDiceRandomSide() {
```
Testing is not random, rename test method to testRollDice1d10

Dyrk84 committed:
DH-9
New method added: chosenNumber() -Now the code take just the correct result from scanner.
Now if the champion kill the monster, the monster can't do counterattack.
Removed the unnecessary empty lines.
Renamed the dice tested methods.
Renamed in characters classes the value numOfDices to championNumOfDices and monsterNumOfDices.
I did a little mess in TestDice class :)

gergely-szanto-epam commented on src/main/java/hu/dungeonhunter/Main.java:
```diff
@@ -0,0 +1,40 @@
+        fight.start();
+        while (fight.nextTurn()) {
+            System.out.println("\nChoose one of the following actions:");
+            System.out.println("1. Attack");
```
You are printing the menu here and at row 36 as well. This will result lots of duplications when the menu is extended with more menu points. Let's create a printMenu() that prints this and call this method here and in row 36.

gergely-szanto-epam commented on src/main/java/hu/dungeonhunter/characters/Champion.java:
```diff
@@ -0,0 +1,35 @@
+    public Champion() {
+        this.championHP = START_HP;
+        this.championMaxDamage = 6;
+        this.championNumOfDices = 1;
```
To print champion data, you can do this here:
```
System.out.println("Your champion have " + getChampionHP() + " HP and can do " +
                getNumOfDices() + "-" + getNumOfDices() * getChampionMaxDamage() +
                " damage.");
```

So this will be printend automatically when you say new Champion(). Whatever is inside the class initializer method will be executed when the class is instantiated.

Dyrk84 replied:
I did, but dont working, all way give back 0 value.

after run, print this:
"Your champion have 0 HP and can do 0-0 damage."

gergely-szanto-epam replied:
Yes, my bad. Getters won't work here yet, because we're not finsihed with the initialization at this point.
Try this:
```
        System.out.println("Your champion have " + championHP + " HP and can do " +
            championNumOfDices + "-" + championNumOfDices * championMaxDamage +
            " damage.");
```

Dyrk84 committed:
DH-9
New printer methods:
-yourChooseIsNotAppropriate()
-printMenu()

Dyrk84 committed:
DH-9
Start method removed, and added new initializer method in the Champion and Monsters constructor.

gergely-szanto-epam commented on src/test/java/hu/dungeonhunter/TestDice.java:
```diff
@@ -0,0 +1,132 @@
+            }
+        } while (true);
+    }
+//// nem tudom hogyan kell a következő 4 tesztben metódus végeredményére hivatkozni, de már nem akarom kitörölni
```
To test victory check you need to introduce a new variable.
I would add a boolean variable to class level for both Champion and Monster, call it victory and give it false value.
In the fight class create a method: setVictory. This should set the victory variable true for a fighter, if the opponent has 0 or less hp.
You should add setVictory to both championAttack and monsterAttack methods, so the victory can be checked after each attack with the getVictory method.

To test you simply set both fighters hp, then call setVictory and assert that the getVictory returns expected value for both fighters (e.g: champion has 0 hp -> monser.getVictory returns true)

Dyrk84 committed:
DH-9
4 unit tests added: championVictory positive and negative test, and monsterVictory positive and negative test
I changed a little bit on the code to run the tests rightly in next classes: Fight, Champion, Monsters

gergelyszanto commented on src/test/java/hu/dungeonhunter/TestDice.java:
```diff
@@ -0,0 +1,146 @@
+        } while (true);
+    }
+
+    // nem tudom hogyan kell a következő 4 tesztben metódus végeredményére hivatkozni, de már nem akarom kitörölni
```
De tudod :D

gergelyszanto commented on src/test/java/hu/dungeonhunter/TestDice.java:
```diff
@@ -0,0 +1,146 @@
+    public void championVictoryPositiveTest() {
+        Champion champion = new Champion();
+        Monsters monster = new Monsters();
+        int championHP = 1;
+        int monsterHP = -1;
```
This is overkill. No need to create variable if it's going to be used only one time (setters).

gergelyszanto commented on src/test/java/hu/dungeonhunter/TestDice.java:
```diff
@@ -0,0 +1,146 @@
+        int championHP = 1;
+        int monsterHP = -1;
+        boolean result;
+        champion.setChampionHP(championHP);
+        monster.setMonsterHP(monsterHP);
```
Simply say: 
```suggestion
        champion.setChampionHP(1);
        monster.setMonsterHP(-1);
```

Dyrk84 committed:
DH-9
added 2 TODO
removed Victory()
